cmd/cluster/core/destroy.go

```diff
@@ -99,7 +100,7 @@ func DestroyCluster(ctx context.Context, hostedCluster *hyperv1.HostedCluster, o
 		if err := c.Patch(ctx, hostedCluster, client.MergeFrom(original)); err != nil {
 			if apierrors.IsNotFound(err) {
 				o.Log.Info("Hosted cluster not found, skipping finalizer update", "namespace", o.Namespace, "name", o.Name)
-			} else {
+			} else if !strings.Contains(err.Error(), "no new finalizers can be added if the object is being deleted") {
```

Comment:
Why not just skip adding the finalizer if the DeletionTimestamp is non-zero?

Reply:
Because this also works if the hostedcluster gets deleted after we fetched it and before we make the API call to add the finalizer